### PR TITLE
Switch CI to persistent clusters with containers

### PR DESCRIPTION
**Pull request by sunkuamzn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sunkuamzn** [comment]
bot:aws:retest

**Comment by sunkuamzn** [comment]
bot:aws:retest

File: .ci/aws/Jenkinsfile [hunk 1]
```diff
@@ -90,38 +66,26 @@ def run_test_orchestrator_once(run_name, build_tag, os, instance_type, instance_
...
-                        returnStdout: true
-                )
 
-    def cluster_name = sh(
-                        script: "echo '${build_tag.take(28)}-${os.take(10)}-${instance_type}-'${get_random_string(8)} | tr -d '.\\n'",
-                        returnStdout: true
-                     )
+    def buildNumber = env.BUILD_NUMBER as int
+    def cluster_name = get_persistent_cluster_name(build_tag, os, instance_type)
+    def args = "--os ${os} --container-os ${container_os} --odcr ${odcr} --instance-type ${instance_type} --instance-count ${instance_count} --region ${region} --cluster-name ${cluster_name} ${addl_args} --junit-xml outputs/${cluster_name}-PR${buildNumber}-${os}-${container_os}.xml"
```
Comment (a-szegel) [inline .ci/aws/Jenkinsfile:76]: Do you need all of these arguments?  ODCR?

**Review by aws-nslick — APPROVED** [review]

File: .ci/aws/Jenkinsfile [hunk 2]
```diff
@@ -229,28 +225,28 @@ pipeline {
...
+                    stages["4_p5_alinux2"] = get_test_stage_with_lock_container("4_p5_alinux2", env.BUILD_TAG, p3_p4_p5_base_os, "alinux2", "p5.48xlarge", p5_region, p5_lock_label, num_instances, p5_odcr, p5_addl_args)
+                    stages["4_p5_ubuntu2004"] = get_test_stage_with_lock_container("4_p5_ubuntu2004", env.BUILD_TAG, p3_p4_p5_base_os, "ubuntu2004", "p5.48xlarge", p5_region, p5_lock_label, num_instances, p5_odcr, p5_addl_args)
+                    stages["4_p5_ubuntu2204"] = get_test_stage_with_lock_container("4_p5_ubuntu2204", env.BUILD_TAG, p3_p4_p5_base_os, "ubuntu2204", "p5.48xlarge", p5_region, p5_lock_label, num_instances, p5_odcr, p5_addl_args)
 
                     // g4dn tests
-                    stages["4_g4dn_ubuntu2204"] = get_test_stage_with_lock("4_g4dn_ubuntu2204", env.BUILD_TAG, "ubuntu2204", "g4dn.12xlarge", g4dn_region, g4dn_lock_label, num_instances, g4dn_odcr, g4dn_addl_args)
+                    stages["4_g4dn_ubuntu2204"] = get_test_stage_with_lock_persistent("4_g4dn_ubuntu2204", env.BUILD_TAG, "ubuntu2204", "g4dn.12xlarge", g4dn_region, g4dn_lock_label, num_instances, g4dn_odcr, g4dn_addl_args)
```
Comment (a-szegel) [inline .ci/aws/Jenkinsfile:243]: Can we run everything through the container path, and clean up the Jenkinsfile?

File: .ci/aws/Jenkinsfile [hunk 3]
```diff
@@ -199,26 +188,33 @@ pipeline {
...
-                    def base_args = "${efa_installer} ${nccl_version} ${timeout} ${cluster_type} ${test_target} ${test_type} ${build_type} ${pr_num} ${test_list} ${nccl_test_iter}"
+
+                    def persistent_manual_cluster_addl_args = " --keep-cluster --skip-fixture-setup --skip-health-checks --use-existing-installer --cleanup-pf-directory"
+                    def container_addl_args = " --test-in-containers-on-ec2"
+
+                    def base_args = "${efa_installer} ${nccl_version} ${timeout} ${cluster_type} ${test_target} ${test_type} ${build_type} ${pr_num} ${nccl_test_iter} ${persistent_manual_cluster_addl_args}"
 
                     def num_instances = 4
                     def p3dn_lock_label = "p3dn-1-4node"
                     def p3dn_region = "ap-northeast-1"
```
Comment (a-szegel) [inline .ci/aws/Jenkinsfile:201]: I think you can reduce the number of arguments here drastically... getting rid of ODCR, and region ;)

**Review by a-szegel — CHANGES_REQUESTED** [review]
Initial review with some high level comments

**Pull request merged**